src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1169,6 +1178,16 @@ void RaftPart::cleanupSnapshot() {
     status_ = Status::RUNNING;
 }
 
+bool RaftPart::needToCleanWal() {
+    for (auto& host : hosts_) {
+        std::lock_guard<std::mutex> g(raftLock_);
```

Comment:
hosts_ should be protected by raftLock_

Reply:
My bad. Hold on please, there is another place to update.